app/models/dinner.rb

```diff
@@ -1,4 +1,20 @@
 class Dinner < ActiveRecord::Base
   validates :email, email: true, case_sensitive: false
   validates :email, :dinner_name, :date, :time, :location, presence: true
+
+  before_save :generate_unique_url
+
+  def custom_url
+    "http://localhost:3000/dinners/#{self.unique_url}"
```

Comment:
Don't hard code `localhost` like this!  The model shouldn't know anything about the other levels of the application, _especially_ routing.